### PR TITLE
Add PT 1.7.1 support to smddp

### PR DESCRIPTION
**Pull request by ChaiBapchya**
Don't merge until PT1.7.1-cu11 training DLC released with SMDDP support
*Issue #, if available:*

*Description of changes:*
Allowlist 1.7.1 PT framework version as a supported SMDDP framework version

*Testing done:*
Ongoing

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ChaiBapchya** [comment]
Allow-list of PT 1.7 can only be done once SM Python SDK has the mapping of framework_version to appropriate DLC image URI
```
PyTorch(framework_version='1.7.0.', ..)
```
to
https://github.com/aws/deep-learning-containers/releases/tag/v1.0-pt-1.7.1-tr-py36

**Comment by ChaiBapchya** [comment]
@saimidu @metrizable @ChoiByungWook FYI

cc @rondogency

File: tests/unit/test_fw_utils.py [hunk 1]
```diff
@@ -632,6 +632,7 @@ def test_validate_smdataparallel_args_not_raises():
         (None, None, None, None, smdataparallel_disabled),
         ("ml.p3.16xlarge", "tensorflow", "2.3.1", "py3", smdataparallel_enabled),
         ("ml.p3.16xlarge", "pytorch", "1.6.0", "py3", smdataparallel_enabled),
+        ("ml.p3.16xlarge", "pytorch", "1.7.0", "py3", smdataparallel_enabled),
```
Comment (ChoiByungWook) [inline tests/unit/test_fw_utils.py:635]: Should this be 1.7.1?

**Comment by ChoiByungWook** [comment]
Pending: https://github.com/aws/sagemaker-python-sdk/pull/2185

**Comment by ChoiByungWook** [comment]
Adding changes in this PR into: https://github.com/aws/sagemaker-python-sdk/pull/2185

When updating the PyTorch 1.7.1 configuration, the integration test for Data Parallel fail due to the conditional mapping for supported versions and the conftest using the latest version.

This will need to be decoupled in the Python SDK tests.

**Pull request closed without merging**